src/analytics/lambdas/scan-metadata-workflow/store-metadata-into-ddb.ts

```diff
@@ -244,15 +344,74 @@ function convertToDDBList(inputString?: string) {
   return listData;
 }
 
-function getCommonData() {
-  return {
-    displayName: '',
-    description: '',
-    hasData: true,
-    updateAt: Date.now(),
-    createAt: Date.now(),
-    operator: '',
-    deleted: false,
-    ttl: (Math.floor(Date.now() / 1000) + 7 * 24 * 60 * 60),
-  };
-}
\ No newline at end of file
+function convertValueEnumToDDBList(inputString?: string) {
+  let listData: any[] = [];
+  if (inputString) {
+    listData = inputString.split('#').map(item => {
+      const lastUnderscoreIndex = item.lastIndexOf('_');
+      const value = item.substring(0, lastUnderscoreIndex);
+      const count = parseInt(item.substring(lastUnderscoreIndex + 1));
+      return { value, count };
+    });
+  }
+  return listData;
+}
+
+async function batchGetDDBItems(allKeys: any[]) {
+  const batchSize = 50;
+  const batchResults: any[] = [];
+  for (let i = 0; i < allKeys.length; i += batchSize) {
+    const currentBatchKeys = allKeys.slice(i, i + batchSize);
+    const request: BatchGetCommandInput = {
+      RequestItems: {
+        [ddb_table_name]: {
+          Keys: currentBatchKeys,
+        },
+      },
+    };
+    const command = new BatchGetCommand(request);
+    try {
+      const response: BatchGetCommandOutput = await ddbDocClient.send(command);
+      batchResults.push(...response.Responses![ddb_table_name]);
+    } catch (error) {
+      console.error(error);
+      return null;
+    }
+  }
+  return batchResults;
+}
+
+async function getExistingItemsFromDDB(appId: string, redshiftTableName: string) {
+  const distinctIdAndMonthSql = `SELECT distinct id, month FROM ${appId}.${redshiftTableName};`;
+  const idAndMonthResponse = await queryMetadata(distinctIdAndMonthSql);
```

Comment:
What's the limit(rows?, size?) of single request to query from Redshift?

Reply:
refactor common method getStatementResult in src/analytics/lambdas/redshift-data.ts
to cover large number records.